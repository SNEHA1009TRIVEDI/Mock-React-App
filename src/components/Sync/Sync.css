.body-padding{
    padding-top: 500px;
    
}
.button{
    border-radius: 25px;  
    background-color: #4C63AA;  
    color:#FFFFFF;  
    text-align :center;  
    font-family:arial, helvetica, sans-serif;  
    padding: 5px 10px 10px 10px;  
    font-weight:bold;  
    width:100px;  
    height:30px;  
     
}