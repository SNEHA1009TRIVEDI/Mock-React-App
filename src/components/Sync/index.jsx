import React from 'react';
import Header from '../Header';
import {useNavigate} from "react-router-dom"
import {HOME_ROUTE} from '../../constants/routes';
import './Sync.css';
const Sync = () => {
    const navigate = useNavigate();
    const handleSync = () => {
        navigate(`${HOME_ROUTE}`);
    }


    return (
        <>
        <Header/>
        <div className='body-padding'>
        <h1 >:((</h1>
        <h2 className="text"> seems a bit empty in here...</h2>

        <button className='button' onClick={handleSync}>Sync</button>
        </div>
        </>
      
        
    );
    }

// export the component
export default Sync;

