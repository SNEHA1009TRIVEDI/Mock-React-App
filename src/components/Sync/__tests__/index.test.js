import { screen, render } from "@testing-library/react";
import React from "react";
import Sync from "..";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  // 3- Import non-mocked library and use other functionalities and hooks
  ...jest.requireActual("react-router-dom"),

  // 4- Mock the required hook
  useNavigate: () => mockedUsedNavigate,
}));
describe("Sync", () => {
  it("should renders without crashing ", () => {
    const { asFragment } = render(<Sync />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should should navigate to home page ", async () => {
    render(
      <BrowserRouter>
        <Sync />
      </BrowserRouter>
    );

    // Testing navigating using the button
    expect(screen.getByText("Sync")).toBeInTheDocument();

    await userEvent.click(screen.getByText("Sync"));
    expect(mockedUsedNavigate).toHaveBeenCalledWith("/record");
  });
});
