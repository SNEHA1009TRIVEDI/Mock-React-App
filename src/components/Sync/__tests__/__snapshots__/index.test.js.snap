// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sync should renders without crashing  1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <h1
      class="heading"
    >
      My 
      <strong>
        Record
      </strong>
       Shelf
    </h1>
  </header>
  <div
    class="body-padding"
  >
    <h1>
      :((
    </h1>
    <h2
      class="text"
    >
       seems a bit empty in here...
    </h2>
    <button
      class="button"
    >
      Sync
    </button>
  </div>
</DocumentFragment>
`;
